src/azure-cli/azure/cli/command_modules/appservice/custom.py

```diff
@@ -1810,13 +1810,18 @@ def list_function_app_runtimes(cmd, os_type=None):
     return {WINDOWS_OS_NAME: windows_stacks, LINUX_OS_NAME: linux_stacks}
 
 
-def list_flex_function_app_runtimes(cmd, location, runtime):
-    runtime_helper = _FlexFunctionAppStackRuntimeHelper(cmd, location, runtime)
-    runtimes = [r for r in runtime_helper.stacks if runtime == r.name]
-    if not runtimes:
-        raise ValidationError("Runtime '{}' not supported for function apps on the Flex Consumption plan."
-                              .format(runtime))
-    return runtimes
+def list_flex_function_app_runtimes(cmd, location, runtime, ignore_error=False):
+    try:
+        runtime_helper = _FlexFunctionAppStackRuntimeHelper(cmd, location, runtime)
+        runtimes = [r for r in runtime_helper.stacks if runtime == r.name]
+        if not runtimes:
+            raise ValidationError("Runtime '{}' not supported for function apps on the Flex Consumption plan."
+                                  .format(runtime))
+        return runtime_helper.stacks
+    except Exception:
+        if ignore_error:
+            return []
+        raise  # Rethrow the caught exception
```

Comment:
Why are we adding this? If we throw the validation exception, we shouldn't just ignore it. This command was meant to only work for a specific runtime. That's why we throw the ValidationError.

Reply:
Detail requests on certain staging locations throws bad request. Will be returning empty/null for that situation.